src/nimony/sem.nim

```diff
@@ -3822,12 +3849,19 @@ proc semWhen(c: var SemContext; it: var Item) =
     producesVoid c, info, it.typ
 
 proc semCaseOfValue(c: var SemContext; it: var Item; selectorType: TypeCursor;
-                    seen: var seq[(xint, xint)]) =
+                    seen: var seq[(xint, xint)]; isString: bool) =
```

Comment:
Instead duplicate this proc so that it takes `seen: var HashSet[string/StrId]` as it needs to check for duplicate string literals too.